frontend/src/reducer/notificationList.js

```diff
@@ -0,0 +1,78 @@
+import { uniqBy } from 'lodash'
+import {
+  ADD,
+  NOTIFICATION,
+  NOTIFICATION_LIST,
+  SET,
+  UPDATE
+} from '../action-creator.sync.js'
+
+const defaultNotificationList = [{
+  author: '',
+  content: '',
+  created: '',
+  icon: '',
+  id: 0,
+  read: false,
+  type: '',
+  url: '',
+  workspace: ''
+}]
+
+export const getNotificationFromTLM = (eventData, contentData) => {
+  let typeIcon, type, url
+
+  if (!/comment/.test(eventData.event_type) && /content.*created/.test(eventData.event_type)) {
+    typeIcon = 'fa-magic'
+    type = 'content.created'
+    url = `/ui/workspaces/${contentData.workspace.workspace_id}/contents/${contentData.content.content_type}/${contentData.content.content_id}`
+  } else if (/content.*modified/.test(eventData.event_type)) {
+    if (contentData.content.current_revision_type === 'status-update') {
+      typeIcon = 'fa-random'
+      type = 'status.modified'
+    } else {
+      typeIcon = 'fa-history'
+      type = 'content.modified'
+    }
+    url = `/ui/workspaces/${contentData.workspace.workspace_id}/contents/${contentData.content.content_type}/${contentData.content.content_id}`
+  } else if (eventData.event_type === 'workspace_member.created') {
+    typeIcon = 'fa-user-o'
+    type = 'member.created'
+    url = `/ui/workspaces/${contentData.workspace.workspace_id}/dashboard`
+  } else return null
+
+  return {
+    author: contentData.author.public_name,
+    content: contentData.content ? contentData.content.label : '',
+    created: eventData.created,
+    icon: typeIcon,
+    id: eventData.event_id,
+    read: eventData.read,
+    type: type,
+    url: url,
+    workspace: contentData.workspace.label
+  }
+}
+
+export default function notificationList (state = defaultNotificationList, action) {
+  switch (action.type) {
+    case `${SET}/${NOTIFICATION_LIST}`: {
+      const notificationList = action.notificationList
+        .map(no => getNotificationFromTLM(no, no.fields))
+        .filter(no => no !== null)
+        .reverse()
+      return uniqBy(notificationList, 'id')
+    }
+
+    case `${ADD}/${NOTIFICATION}`: {
+      // INFO - 2020-07-22 - GB - The event and the content data are in the same place, that's why we send the same constant
+      return uniqBy([getNotificationFromTLM(action.notification, action.notification), ...state], 'id')
```

Comment:
I think the naming of something can be improved to avoid having to give these strange parameters.
I tried a bit to find something but I couldn't. The only thing that is bugging me is that the list from `${SET}/${NOTIFICATION_LIST}` is from an api endpoint and the item from `${ADD}/${NOTIFICATION}` is from a TLM.

Since the endpoint return "messages" and, from a backend point of view, messages are TLM, I don't know how to fix because it seems logical. Yet it requires 2 different behavior...

Or maybe this show a design flaw at a higher level about how we build TLM objects.

Anyway, I require nothing about this in my review, I'm just giving my thoughts on that

Reply:
oh and, about your comment, we usually say that we "give" parameters to a function, not "send" them

Reply:
I agree with you 100%, that's why I saw the need to put the comment. One solution I thought would be to change the nomenclature in LiveMessageManager to keep the same as the API, this implies updating all the code (whenever we use "data", we would use "data.fields"). It seems coherent to me, what do you think?

Reply:
Yes, that seems to be a good idea.
It can be a specific issue if it gets too complicated or if the refactor gets too big though

Reply:
Changed